specs/clickhouse.spec

```diff
@@ -0,0 +1,323 @@
+%define _loc_prefix       %{_prefix}/local
+%define _loc_exec_prefix  %{_loc_prefix}
+%define _loc_bindir       %{_loc_exec_prefix}/bin
+%define _loc_libdir       %{_loc_exec_prefix}/%{_lib}
+%define _loc_libdir32     %{_loc_exec_prefix}/%{_lib32}
+%define _loc_libdir64     %{_loc_exec_prefix}/%{_lib64}
+%define _loc_libexecdir   %{_loc_exec_prefix}/libexec
+%define _loc_sbindir      %{_loc_exec_prefix}/sbin
+%define _loc_bindir       %{_loc_exec_prefix}/bin
+%define _loc_datarootdir  %{_loc_prefix}/share
+%define _loc_includedir   %{_loc_prefix}/include
+%define _loc_mandir       %{_loc_datarootdir}/man
+%define _rpmstatedir      %{_sharedstatedir}/rpm-state
+%define _pkgconfigdir     %{_libdir}/pkgconfig
+
+%define __ldconfig        %{_sbin}/ldconfig
+%define __service         %{_sbin}/service
+%define __touch           %{_bin}/touch
+%define __chkconfig       %{_sbin}/chkconfig
+%define __updalt          %{_sbindir}/update-alternatives
+%define __useradd         %{_sbindir}/useradd
+%define __groupadd        %{_sbindir}/groupadd
+%define __getent          %{_bindir}/getent
+%define __systemctl       %{_bindir}/systemctl
+
+################################################################################
+
+%define service_user       clickhouse
+%define service_group      clickhouse
+%define service_data_dir   %{_sharedstatedir}/clickhouse
+%define service_log_dir    %{_logdir}/clickhouse-server
+
+################################################################################
+
+Summary:           Yandex ClickHouse DBMS
+Name:              clickhouse
+Version:           18.16.1
+Release:           0%{?dist}
+License:           APL 2.0
+Group:             Applications/Databases
+URL:               https://clickhouse.yandex
+
+Source:            %{name}-%{version}.tar.bz2
+
+BuildRoot:         %{_tmppath}/%{name}-%{version}-%{release}-root-%(%{__id_u} -n)
+
+BuildRequires:     centos-release-scl devtoolset-7
+BuildRequires:     cmake3 openssl-devel libicu-devel libtool-ltdl-devel
+BuildRequires:     unixODBC-devel readline-devel
+
+Requires:          openssl libicu libtool-ltdl unixODBC readline
+
+%if 0%{?rhel} >= 7
+Requires(pre):     shadow-utils
+Requires(post):    systemd
+Requires(preun):   systemd
+Requires(postun):  systemd
+%else
+Requires(pre):     shadow-utils
+Requires(post):    chkconfig
+Requires(preun):   chkconfig
+Requires(preun):   initscripts
+Requires(postun):  initscripts
+%endif
+
+Provides:          %{name} = %{version}-%{release}
+
+################################################################################
+
+%description
+ClickHouse is an open-source column-oriented database management
+system that allows generating analytical data reports in real time.
+
+################################################################################
+
+%package client
+
+Summary:           ClickHouse client binary
+Group:             Applications/Databases
+
+Requires:          %{name}-server = %{version}-%{release}
```

Comment:
Why do we need to install `clickhouse-server` along with `clickhouse-client`?

Reply:
If to be honest, I have no idea — this spec based on [spec](https://github.com/Altinity/clickhouse-rpm/blob/master/src/clickhouse.spec.in#L56) file created by Altinity and fully compatible with packages based on it. In their spec, `client` package requires `server` package.